gaek/ndb_json.py

```diff
@@ -109,28 +134,52 @@ def encode_basevalue(obj):
   ndb.MetaModel: encode_model,
   ndb.Query: encode_generator,
   ndb.QueryIterator: encode_generator,
-  ndb.Key: encode_key,
+  ndb.Key: encode_key_as_entity,
   ndb.Future: encode_future,
   datetime.date: encode_datetime,
   datetime.datetime: encode_datetime,
   time.struct_time: encode_generator,
   types.ComplexType: encode_complex,
   ndb.model._BaseValue: encode_basevalue,
-  
 }
 
 
 class NdbEncoder(json.JSONEncoder):
   """Extend the JSON encoder to add support for NDB Models."""
 
+  _ndb_type_encoding=NDB_TYPE_ENCODING.copy()
+
+  def __init__(self, **kwargs):
+    ndb_keys_as_entities = kwargs.pop('ndb_keys_as_entities', False)
+    ndb_keys_as_pairs = kwargs.pop('ndb_keys_as_pairs', False)
+    ndb_keys_as_urlsafe = kwargs.pop('ndb_keys_as_urlsafe', False)
+
+    # Validate that only one of three flags is True
+    if ((ndb_keys_as_entities and ndb_keys_as_pairs)
+        or (ndb_keys_as_entities and ndb_keys_as_urlsafe)
+        or (ndb_keys_as_pairs and ndb_keys_as_urlsafe)):
+      raise NDBJsonInvalidArgumentsError(
```

Comment:
Simply raising `ValueError` would be acceptable here since it's a usage issue and not really an internal error.